echopop/tests/test_kriging_biomass_apportionment.py

```diff
@@ -0,0 +1,878 @@
+        'proportion_aged_weight_all': [ 0.10 , 0.20 ] ,
+        'proportion_aged_weight_adult': [ 0.05 , 0.10 ] ,
+        'proportion_unaged_weight_all': [ 0.90 , 0.80 ] ,
+        'proportion_unaged_weight_adult': [ 0.45 , 0.40 ]
+    } )
+
+    
+    ### Evaluate for later comparison 
+    eval_aged_sex_proportions = compute_aged_sex_weight_proportions( test_proportions_weight_length_age_sex ,
+                                                                     test_aged_proportions )
+    ###--------------------------------
+    ### Expected outcomes
+    ###--------------------------------
+    ### `eval_aged_sex_proportions`
+    # ---- Expected dimensions
+    expected_dimensions = tuple( [ 4 , 6 ] )
+    # ---- Expected dataframe output
+    expected_output = pd.DataFrame( {
+        'stratum_num': np.repeat( [ 0.0 , 1.0 ] , 2 ).astype( np.int64 ) ,
+        'sex': np.tile( [ 'female' , 'male' ] , 2 ).astype( object ) ,
+        'proportion_aged_weight_all': [ 0.10 , 0.10 , 0.20 , 0.20 ] ,
+        'proportion_aged_weight_adult': [ 0.05 , 0.05 , 0.10 , 0.10 ] ,
+        'proportion_weight_all': [ 0.050 , 0.050 , 0.040 , 0.040 ] ,
+        'proportion_weight_adult': [ 0.035 , 0.040 , 0.024 , 0.036 ] 
+    } )
+
+    #----------------------------------
+    ### Run tests: `compute_index_aged_weight_proportions`
+    #----------------------------------
+    ### `eval_aged_sex_proportions`
+    # ---- Shape
+    assert eval_aged_sex_proportions.shape == expected_dimensions
+    # ---- Datatypes
+    assert np.all( eval_aged_sex_proportions.dtypes == expected_output.dtypes )
+    # ---- Dataframe equality
+    assert np.all( eval_aged_sex_proportions == expected_output )
+
+def test_distribute_aged_weight_proportions( ):
+
+    ### Mock data for `proportions_weight_length_age_sex`
+    test_proportions_weight_length_age_sex = pd.DataFrame( 
+        {
+            'stratum_num': np.repeat( [ 0.0 , 1.0 ] , 2 ) ,
+            'sex': np.tile( [ 'female' , 'male' ] , 2 ) ,
+            'proportion_weight_sex_all': [ 0.50 , 0.50 , 0.25 , 0.50 ] ,
+            'proportion_weight_sex_adult': [ 0.50 , 0.25 , 0.00 , 0.25 ]
+        } 
+    )
+
+    ### Mock data for `aged_sex_proportions`
+    test_aged_sex_proportions = pd.DataFrame( {
+        'stratum_num': np.repeat( [ 0.0 , 1.0 ] , 2 ).astype( np.int64 ) ,
+        'sex': np.tile( [ 'female' , 'male' ] , 2 ).astype( object ) ,
+        'proportion_aged_weight_all': [ 0.10 , 0.10 , 0.20 , 0.20 ] ,
+        'proportion_aged_weight_adult': [ 0.05 , 0.05 , 0.10 , 0.10 ] ,
+        'proportion_weight_all': [ 0.050 , 0.050 , 0.040 , 0.040 ] ,
+        'proportion_weight_adult': [ 0.035 , 0.040 , 0.024 , 0.036 ] 
+    } )
+
+    ### Evaluate for later comparison 
+    eval_distributed_aged_weight_proportions = distribute_aged_weight_proportions( test_proportions_weight_length_age_sex ,
+                                                                                   test_aged_sex_proportions )
```

Comment:
I am confused about what `proportions_weight_length_age_sex` and `aged_sex_proportions` are at the input of this function. I think this is because the proportions do not sum to 1 in the columns I expected them to, so I can't tell if `proportions_weight_length_age_sex` is the weight proportion of each length and age bin WITHIN each sex, or if it is the weight proportion of each sex, length, and age bin. I have similar question about `aged_sex_proportions`... 

Maybe it would be the best if there is a table that maps these quantities to what is in the documentation?